src/main/java/gift/controller/ProductController.java

```diff
@@ -0,0 +1,64 @@
+package gift.controller;
+
+import gift.dto.ProductRequest;
+import gift.dto.ProductResponse;
+import gift.service.ProductService;
+import java.util.List;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/products")
+public class ProductController {
+
+  private final ProductService productService;
+
+  public ProductController(ProductService productService) {
+    this.productService = productService;
+  }
+
+  @GetMapping
+  public ResponseEntity<List<ProductResponse>> getProducts() {
+    return ResponseEntity.ok(productService.getAllProducts());
+  }
+
+  @GetMapping("/{productId}")
+  public ResponseEntity<ProductResponse> getProduct(@PathVariable Long productId) {
+    return ResponseEntity.ok(productService.getProductById(productId));
+  }
+
+  @PostMapping
+  public ResponseEntity<ProductResponse> addProduct(@RequestBody ProductRequest request) {
+    ProductResponse response = productService.addProduct(request);
+    return ResponseEntity.status(201).body(response);
```

Comment:
httpStatusCode를 하드코딩하셨네요. 👀

이미 정의된 springframework의 `HttpStatus` enum을 활용해보면 좋을 것 같아요. 😃

```suggestion
    return new ResponseEntity(response, HttpStatus.CREATED);
```

추가적으로 ResponseEntity는 편의를 위한 **정적 팩토리 메서드**들을 제공하고있는데, 적절히 활용해보면 좋을 것 같아요. 😃 (이미 위에서 ResponseEntity의 ok(), notFound()와 같은  정적 팩토리 메서드를 잘 활용해주네요?)

<img width="780" alt="image" src="https://github.com/user-attachments/assets/0ebf33ae-f717-4ee7-ac10-525d84c9487c" />


이미 알고계실 수도 있지만 이번 기회에 [정적 팩토리 메서드](https://tecoble.techcourse.co.kr/post/2020-05-26-static-factory-method/) 개념에 대해서 공부해보면 좋을 것 같아요. 😉